Add FAQ page, and other doc tweaks (#145)
blockprotocol/blockprotocol: site/src/_pages/docs/4_faq.mdx

## Schemas, vocabularies and the semantic web

### What is the semantic web?

The goal of the [Semantic Web](https://en.wikipedia.org/wiki/Semantic_Web) is to make internet data machine-readable.

It involves making sure there is data on web pages which can be parsed by machines, in order to determine the entities described, and their links to other entities elsewhere on the web.

### How does the Block Protocol advance the semantic web?

The Block Protocol requires that the data passed between blocks and applications is in the form of entities conforming to a defined structure: a schema.

This structure can then be used to include data on web pages which describes the entities in a machine-readable way.

For example, where a page contains a block displaying a movie, machine-readable data would also be included describing the various properties of the movie (e.g. `releaseDate`), as well as linking to other pages which describe entities linked to it - e.g. its `director`.

> One way of representing entities in a machine-readable way is in [JSON-LD](https://json-ld.org/). which describes entities and their link to other entities).Here’s an example of such a representation, taken from [the JSON-LD homepage](https://json-ld.org/):

Reviewer: ```suggestion
One way of representing entities in a machine-readable way is in [JSON-LD](https://json-ld.org/) which describes entities and their link to other entities). Here’s an example of such a representation, taken from the JSON-LD homepage:
```

```json
{
  "@context": "https://json-ld.org/contexts/person.jsonld",
  "@id": "http://dbpedia.org/resource/John_Lennon",
  "name": "John Lennon",
  "born": "1940-10-09",
  "spouse": "http://dbpedia.org/resource/Cynthia_Lennon"
}
```

### Is the Block Protocol an ontology of the real-world?

No, and _for the most part_ the Block Protocol itself does not define the structure of entities passed between blocks and applications. Instead it specifies _how_ entities should be defined, transmitted and updated.

The Block Protocol specifies that [JSON Schema](https://json-schema.org/) should be used to describe the properties of an entity and expected types of property values - not what those properties should be.

> You can use existing [Schema.org](http://Schema.org) schemas, or can create new schemas through the [B](http://BlockProtocol.org)lock Protocol website, once logged in. The schema creator on the Block Protocol website aims to make it easier for users to define the different types of data structures that their blocks will work with.

Reviewer: ```suggestion
You can use existing [Schema.org](https://schema.org) schemas, or can create new schemas through the [Block Protocol](https://blockprotocol.org/) website, once logged in. The schema creator on the Block Protocol website aims to make it easier for users to define the different types of data structures that their blocks will work with.
```

> We don’t prescribe canonical types for any _thing_, but do recommend that users link their schemas and properties to [schema.org](http://schema.org) types and properties where possible, to help in making the pages that use their schemas machine-readable. We have included a way of doing so in our schema editor. Mapping different schemas and their properties to one another is a process known as ‘crosswalking’.

Reviewer: ```suggestion
We don’t prescribe canonical types for any _thing_, but do recommend that users link their schemas and properties to [schema.org](https://schema.org) types and properties where possible, to help in making the pages that use their schemas machine-readable. We have included a way of doing so in our schema editor. Mapping different schemas and their properties to one another is a process known as ‘crosswalking’.
```

### How does the Block Protocol relate to schema.org?

> [schema.org](http://schema.org) defines a collection of schemas for use in making the data on web pages machine-readable.

Reviewer: ```suggestion
[schema.org](https://schema.org) defines a collection of schemas for use in making the data on web pages machine-readable.
```

The Block Protocol does not define the structure of entities passed between blocks and applications - it only specifies how they are transmitted and updated.

Implementers of the Block Protocol could use the structure of entities as defined by [schema.org](http://schema.org), if they wished (e.g. a block could be built to render or edit a [schema.org Event)](https://schema.org/Event).

<InfoCardWrapper>

<InfoCard variant="warning" title="Crosswalked?">

Mapping different schemas and their properties to one another is a process known as ‘crosswalking’. It helps machines more easily understand how new schemas fit in to existing knowledge graphs and ontologies.

</InfoCard>

We recommend that users link their schemas and their properties to [schema.org](http://schema.org) types and properties where possible, and have included a way of doing so in our schema editor.

> You can find all [Schema.org](http://Schema.org) definitions on the Block Protocol website already, ready to be ‘crosswalked’ back to.

Reviewer: ```suggestion
You can find all [schema.org](https://schema.org) definitions on the Block Protocol website already, ready to be ‘crosswalked’ back to.
```

</InfoCardWrapper>

### Why does the Block Protocol include a schema editor?

Schemas are central to our vision for the Block Protocol.

We don’t believe that a single set of schemas provided by any one standards organization (even schema.org) can ever perfectly fit all use cases, and as such the ability to create new schemas is important. When doing this, we want to make them as accessible to machines (and people) as possible.

<InfoCardWrapper>

The Block Protocol website includes a schema editor that provides a convenient way to define new entity types used by or with blocks and their embedding applications. These are then hosted persistently and made accessible via the same content delivery network that serves blocks from the Block Protocol Hub, guaranteeing their availability and discoverability.

<InfoCard variant="warning" title="Who can see my schemas?">

All schemas created on the Block Protocol website are currently public. In future we’ll support the creation of private schemas as well.

</InfoCard>

</InfoCardWrapper>

> ### Why aren’t the default [Schema.org](http://Schema.org) definitions sufficient?

Reviewer: ```suggestion
### Why aren’t the default [schema.org](https://schema.org) definitions sufficient?
```

> [Schema.org](http://Schema.org) provides a great base ontology for defining lots of types of ‘things’ out there in the world.

Reviewer: ```suggestion
[schema.org](https://schema.org) provides a great base ontology for defining lots of types of ‘things’ out there in the world.
```

1.  Many times, certain ‘properties’ or data won’t be relevant to your use case, resulting in bloated entity type definitions.
1.  On other occasions you’ll want to store information differently than how it’s set out in the [schema.org](http://schema.org) definition. For example, [schema.org/Person](http://schema.org/Person) defines a person as having a `givenName` and a `familyName` and yet in various cultures this isn’t guaranteed. HASH uses `preferredName` and `legalName` instead - in communications and billing contexts respectively. You can view the HASH ‘Person’ schema at [blockprotocol.org/types/XXXXX](http://blockprotocol.org/types/XXXXX) and see how it crosswalks with the canonical definition of a person provided by Schema.org. This is just one example of how custom schemas can be made to relate back to the core ontology provided by Schema.org.

> 1.  The [Schema.org](http://Schema.org) ontology is slow to change. This is by design, similar to how most standards organizations operate. And while it can move fast when clear universal impetus exists (evidenced by its excellent response to COVID) it ultimately represents a pseudo-centralized model of maintaining a schema registry. The Block Protocol provides the permissionless ability to build atop and extend [schema.org](http://schema.org), unbounded by the processing constraints and imagination of any single working group or organization.

Reviewer: ```suggestion
1.  The [schema.org](https://schema.org) ontology is slow to change. This is by design, similar to how most standards organizations operate. And while it can move fast when clear universal impetus exists (evidenced by its excellent response to COVID) it ultimately represents a pseudo-centralized model of maintaining a schema registry. The Block Protocol provides the permissionless ability to build atop and extend [schema.org](https://schema.org), unbounded by the processing constraints and imagination of any single working group or organization.
```

> 1.  The [Schema.org](http://Schema.org) ontology is slow to change. This is by design, similar to how most standards organizations operate. And while it can move fast when clear universal impetus exists (evidenced by its excellent response to COVID) it ultimately represents a pseudo-centralized model of maintaining a schema registry. The Block Protocol provides the permissionless ability to build atop and extend [schema.org](http://schema.org), unbounded by the processing constraints and imagination of any single working group or organization.

Reviewer: ```suggestion
1.  The [schema.org](https://schema.org) ontology is slow to change. This is by design, similar to how most standards organizations operate. And while it can move fast when clear universal impetus exists (evidenced by its excellent response to COVID) it ultimately represents a pseudo-centralized model of maintaining a schema registry. The Block Protocol provides the permissionless ability to build atop and extend [schema.org](http://schema.org), unbounded by the processing constraints and imagination of any single working group or organization.
```

> 1.  Not everything is included in the [Schema.org](http://Schema.org) ontology. For example, although _animal shelters_ and _pet stores_ exist, _animals_ and _pets_ themselves are missing entirely. This arguably reflects the priorities of Schema.org’s maintainers (notably corporates, and search engines at that).

Reviewer: ```suggestion
1.  Not everything is included in the [schema.org](https://schema.org) ontology. For example, although _animal shelters_ and _pet stores_ exist, _animals_ and _pets_ themselves are missing entirely. This arguably reflects the priorities of Schema.org’s maintainers (notably corporates, and search engines at that).
```

### Why JSON Schema for describing entity structure?

> JSON Schema has a rich [vocabulary for validating data.](https://json-schema.org/draft/2020-12/json-schema-validation.html) We designed the Block Protocol with content management systems and collaborative workspaces in mind, and being able to define precise constraints for data is important for those applications.

Reviewer: ```suggestion
JSON Schema has a rich [vocabulary for validating data](https://json-schema.org/draft/2020-12/json-schema-validation.html). We designed the Block Protocol with content management systems and collaborative workspaces in mind, and being able to define precise constraints for data is important for those applications.
```
